private/pkg/protoencoding/protoencoding.go

```diff
@@ -37,6 +37,17 @@ func NewResolver(fileDescriptors ...protodescriptor.FileDescriptor) (Resolver, e
 	return newResolver(fileDescriptors...)
 }
 
+// NewLazyResolver creates a new Resolver that is constructed from the given
+// descriptors only as needed, if invoked.
+//
+// If there is an error when constructing the resolver, it will be returned by all
+// method calls of the returned resolver.
+func NewLazyResolver(fileDescriptors ...protodescriptor.FileDescriptor) Resolver {
```

Comment:
Is this an optimization to how things worked before? I noticed GetImage changed from `NewResolver` to `NewLazyResolver`.